src/main/java/com/Incubatormsrv/Incubatormsrv/WeatherController.java

```diff
@@ -0,0 +1,17 @@
+package com.Incubatormsrv.Incubatormsrv;
+
+import _Users_luyanda_glucode.com_Documents_GitHub_Incubator_msrv.handler.CurrentWeatherApi;
+import _Users_luyanda_glucode.com_Documents_GitHub_Incubator_msrv.model.CurrentWeather;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.security.auth.callback.Callback;
+
+@RestController
+public class WeatherController implements CurrentWeatherApi {
+
+    @Override
+    public ResponseEntity<CurrentWeather> currentWeatherGet(String q, String numOfDays, String format, String key) {
+        return CurrentWeatherApi.super.currentWeatherGet(q, numOfDays, format, key);
```

Comment:
This isn't correct at all, does this even return anything at all? 

The structure of the code to ensure Single responsibility is achieved. 

You have your controller that implements the API interface generated for Spring language, that makes use of a service wrapper where the service wrapper invokes your retrofit created 3rd party defined interface and makes the service call. Once you have the response from the 3rd party api, it returns it back to the controller and then you have a mapper where the mapper is a separate class to inject into your controller and it takes the 3rd party API response models and maps it to your controller models.